### ISSUE TITLE
[FireFox] Video Stutters in Veetle

### ISSUE THREAD

**Issue by Larryrgs** [uBlock-LLC/uBlock#905]
Watching any video in Veetle.com for a few minutes, The video starts to stutter and CPU spikes.
This problem is also present in ABP but the workaround is to turn off "count filter hits" in options.
Can a similar fix be implemented?

**Comment by gorhill** [comment]
How much is a "few minutes"? I've been watching one for a while now and no issue.

There is no filter hit counters in uBlock.

**Comment by gorhill** [comment]
Also give me an exact URL so that we can look exactly at the same thing. Also, filter lists etc, **all** setings which differs from default. OS. Flash version.

**Comment by Larryrgs** [comment]
try this link for example (happens to me in any link)
http://veetle.com/index.php/channel/view#521135971d84a/7d62989c04d43d65d414ab06dabee4fa

the stuttering (jerkiness) starts happening at about 2 min in. issue disappears when disabling ublock but ads then eventually popup in the video.
Im using all the default settings and filter lists in ublock
Im on Windows 8.1 (64bit)
FF 37 (problem is also present on earlier versions)
flash 17 (problem is also on previous flash)

**Comment by gorhill** [comment]
The site is designed to "sabotage" itself if it can't fetch the ad from the server. Once this happens, it will firehose network requests:

![c](https://cloud.githubusercontent.com/assets/585534/6427527/80ebe9b2-bf51-11e4-9a6b-e51adfb8338b.png)

The count never stops going up. It doesn't look like anything can be done.

**Comment by Larryrgs** [comment]
Thanks for your fast response

**Comment by gorhill** [comment]
I am experimenting... It does look this filter help:

```
||s0.2mdn.net/instream/$important,domain=veetle.com
```

Not sure though... Can you try?

**Comment by harshanvn** [comment]
I tried to reproduce this issue (video stuttering).
But unable to do so. Video plays smoothly. Tried it for greater than 3 minutes.
However, I have dynamic filtering in 3P Deny mode, and could see it blocking all the unnecessary requests. I see count > 6K :)

My Config (W8.1, FF 36, v0.8.9.1,3P, Laptop in Power Saver mode, CPU Freq was set 0.75 GHz, CPU never goes above 14%)

I tried to disable dynamic filtering, but video no longer plays now. It simply says video unavailable. Let me try for another video and see how it goes..Will update here in this post.

**Comment by gorhill** [comment]
Blocking `s0.2mdn.net` seems to help a lot from my side.

**Comment by gorhill** [comment]
Ok, it seems a filter I added a long time ago to fix #374 is actually the problem. Will fix asap.

The page will still try to fire network requests, but it appears this is not enough for the problem. This explains why this could be fixed with ABP by disabling the hit counters, because this probably reduced the overhead introduced for each requests.

**Comment by gorhill** [comment]
You should be able to see an update available for uBlock's filters if you go in _3rd-party filters_ pane. If not, just restart uBlock, then you should be able to see it.

The count in the badge will still go up non-stop, because the amount of network requests blocked is crazy, maybe disabling the badge is a good idea if you are going to watch videos on that site. I am currently at 12K+.

**Comment by Larryrgs** [comment]
Amazing!  Thanks again for your stunning speed

**Comment by harshanvn** [comment]
Yes. That's seems to be issue. Since it got blocked by default through dynamic filtering. I did not see the stutter. However i was able to reproduce the issue, when disabled it.

So, i think blocking <code>s0.2mdn.net</code> should work the people who do not use 3P Deny mode.
Tested URL: http://veetle.com/index.php/channel/view#545c121e5d653

However, one strange thing i observed with this version, the counter does not automatically increment. Once i click popup UI it gets updated.

[will raise a new bug, to track it properly]

**Comment by gorhill** [comment]
> the counter does not automatically increment

It does, but the update is constantly delayed because the count is constantly updated. Updating the badge is considered a low priority task.

By the way, if you use dynamic filtering, it's best to block `s0.2mdn.net` using dynamic filtering, there is much less overhead blocking with dynamic filtering than static filtering, so this should help further lower the overhead of these non-stop network requests.

**Comment by harshanvn** [comment]
> It does, but the update is constantly delayed because the count is constantly updated. Updating the badge is considered a low priority task.

Oh I see the reason. And yes, it is low priority only.

> By the way, if you use dynamic filtering, it's best to block s0.2mdn.net using dynamic filtering

I configured to block all 3P requests <code>3rd-part</code> by default. So, it took care of it :)

**Comment by Betsy25** [comment]
You guys got that 50 shades video playing without stuttering every 4 seconds in Firefox ?
No matter what I tried, latest uBlock or not, FF safe profile or not, it was always stuttering like crazy.
Opening it in IE it plays just fine (but what a bunch of ads)

**Comment by gorhill** [comment]
Actually I don't see the non-stop firing of net requests when using default-deny (which can be set only for that site) and the following rules:

    * * 3p block
    veetle.com ajax.googleapis.com * noop
    veetle.com d3u3s1yur1h76d.cloudfront.net * noop
    # probably you will have to add more if video doesn't play
    # I plan to add ability to 38.102.* eventually
    veetle.com 38.102.162.229 * noop
    veetle.com 38.102.162.242 * noop
    veetle.com 38.102.162.254 * noop

Not sure if more rules are needed for other functionalities, but the videos played just fine as far as I can tell.

@harshanvn "yes, it is low priority only"

I meant among all the things uBlock has to do, updating the badge is not a top priority task. There is nothing to fix.

**Comment by gorhill** [comment]
@Betsy25 did you get the fix?

**Comment by Betsy25** [comment]
@gorhill Yeps I did, both in FF nightly and stable. Same experience, plays 3 seconds, throttler for 6 seconds, play 3 seconds, ...........

**Comment by gorhill** [comment]
Use a dynamic `veetle.com 2mdn.net * block`, less overhead. Also, be sure there is no logging window opened where the blocked requests are logged, this will eat your CPU/memory. Also, any other extensions which maybe do not handle well firehose net requests?

**Comment by Betsy25** [comment]
@gorhill sometimes it plays for 10secs now, otherwise not much change, think this is a Firefox problem, since it happens just as much in safe mode, Firefox stable or nightly, doesn't matter.
Please diregard my case, it's a FF thing. It's playing perfectly fine in IE.

**Comment by gorhill** [comment]
There is something to do on uBlock's side: even if the badge is turned off, there is still a call to update the badge when a request is blocked. I added a check first to call **only** if the badge is turned on, this should remove a chunk of overhead for the current case. I wonder if this will make a different on your side.

I am currently at 10K+, so this would be 10K worht of call to update the badge being skipped.
